public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -460,18 +473,13 @@ export default class LokiLanguageProvider extends LanguageProvider {
         this.labelsCache.set(cacheKey, value);
 
         // Add to label options
-        this.logLabelOptions = this.logLabelOptions.map(keyOption => {
-          if (keyOption.value === key) {
-            return {
-              ...keyOption,
-              children: values.map(value => ({ label: value, value })),
-            };
-          }
-          return keyOption;
-        });
+        this.logLabelOptions = addLabelValuesToOptions(values);
       } catch (e) {
         console.error(e);
       }
+    } else {
+      // Add to label options
```

Comment:
This comment is repeating the code. Perhaps either clarify what "option" means here, or clarify what value is being added here, e.g., the cached vs the uncached one.